openmp/runtime/src/CMakeLists.txt

```diff
@@ -422,6 +422,48 @@ elseif(${LIBOMP_FORTRAN_MODULES})
   set(BUILD_FORTRAN_MODULES True)
 endif()
 
+if(BUILD_FORTRAN_MODULES)
+  if(NOT LIBOMP_FORTRAN_MODULES AND ${LIBOMP_FORTRAN_MODULES_COMPILER} STREQUAL "")
+    message(FATAL_ERROR
+      "Generating OpenMP Fortran modules now requires LIBOMP_FORTRAN_MODULES=ON "
+      "or LIBOMP_FORTRAN_MODULES_COMPILER to be set so the adaptor implementations "
+      "can be compiled and linked into libomp.")
+  endif()
+  target_sources(omp PRIVATE kmp_ftn_support.f90)
+  set_source_files_properties(kmp_ftn_support.f90
+                              PROPERTIES COMPILE_FLAGS "${LIBOMP_CONFIGURED_FFLAGS}")
+  add_dependencies(omp libomp-mod)
+  # Determine Fortran runtime libraries to link
+  # For LLVM flang, we need flang_rt.runtime (newer)
+  set(_LIBOMP_FORTRAN_LIBS "")
+  if(CMAKE_Fortran_COMPILER_ID MATCHES "LLVMFlang" OR CMAKE_Fortran_COMPILER_ID MATCHES "Flang")
+    # Derive paths from Fortran compiler location
+    get_filename_component(_FC_DIR "${CMAKE_Fortran_COMPILER}" DIRECTORY)
+    get_filename_component(_FC_PREFIX "${_FC_DIR}" DIRECTORY)
+
+    # Try to find the flang runtime in clang resource directory
+    # Path: <prefix>/lib/clang/<version>/lib/<target>/libflang_rt.runtime.a
+    file(GLOB _FLANG_RT_CANDIDATES
+      "${_FC_PREFIX}/lib/clang/*/lib/*/libflang_rt.runtime.a"
+    )
+    if(_FLANG_RT_CANDIDATES)
+      list(GET _FLANG_RT_CANDIDATES 0 _FLANG_RT_LIB)
```

Comment:
IIUC this just takes the first libflang_rt.runtime.a. This will likely if compiling more than one flang_rt for another target, or of building libflang_rt.runtime.so instead.